qfieldsync/core/cloud_converter.py

```diff
@@ -128,6 +129,19 @@ def convert(self) -> None:  # noqa: C901
                             self.project.removeMapLayer(layer)
                             continue
                 else:
+                    # Validate filenames and paths before copying
+                    if not is_valid_filename(layer.name()) or not is_valid_path(
+                        layer.source()
+                    ):
```

Comment:
```suggestion
                    if not is_valid_path(str(layer.filename)):
```
I am not sure if we need to run `str`, but the idea is that `is_valid_path` is already using `is_valid_filename`

Reply:
You are right, completely agree in the path validation have validation for the filename!